File: 백준/Silver/14425. 문자열 집합/문자열 집합.kt
fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.`out`.bufferedWriter()

    val (n, m) = br.readLine().split(" ").map {it.toInt()}
    val textSet = HashSet<String>()
    repeat(n){
        textSet.add(br.readLine())
    }

    var count  = 0
    repeat(m){
        if(br.readLine() in textSet){
            count++
        }
    }
    bw.write(count.toString())
    br.close()
    bw.flush()
    bw.close()
}

File: 백준/Silver/14425. 문자열 집합/문자열 집합.py
import sys
input = sys.stdin.readline
N, M = map(int, input().split())
textList = set([input() for _ in range(N)])
cnt = 0

for i in range(M):
    subText = input()
    if subText in textList:
        cnt+=1
#write(str(cnt)+"\n")
print(cnt)